### PR TITLE
Adds logging to Exosuit Console EMP

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request
Adds logging for when the Exosuit Console EMP function is used.

## Why It's Good For The Game
Helps admins retrace players' steps. Includes both an entry to the game.log as well as an admin message. Admin messages include follow/jump links for all involved, while the game.log will list the mech's location coordinates. 

## Changelog
:cl:
admin: Added logging for when an Exosuit Console is used to EMP a mech.
/:cl:

Game.log entry: `GAME: Zxaber/(Ivy Cox) has activated remote EMP on exosuit the �Durand, located at (Research and Development (131, 123, 2)), which is currently without a pilot. `
Admin message:
image

### CODE REVIEW

**Comment by TheVekter** [comment]
IT'S BIG, IT'S HEAVY, IT'S WOOD

File: code/game/mecha/mecha_control_console.dm [hunk 1]
```diff
@@ -41,8 +41,14 @@
 		return
 	if(href_list["shock"])
 		var/obj/item/mecha_parts/mecha_tracking/MT = locate(href_list["shock"])
-		if (istype(MT))
+		if (istype(MT) && MT.chassis)
 			MT.shock()
+			//if(MT.chassis.occupant)
```
Comment (81Denton) [inline code/game/mecha/mecha_control_console.dm]: What's up with the commented out code block?

Comment (zxaber) [reply code/game/mecha/mecha_control_console.dm]: Forgot about that.

File: code/game/mecha/mecha_control_console.dm [hunk 2]
```diff
@@ -41,8 +41,10 @@
 		return
 	if(href_list["shock"])
 		var/obj/item/mecha_parts/mecha_tracking/MT = locate(href_list["shock"])
-		if (istype(MT))
+		if (istype(MT) && MT.chassis)
 			MT.shock()
+			log_game("[key_name(usr)] has activated remote EMP on exosuit [MT.chassis], located at [loc_name(MT.chassis)], which is currently [MT.chassis.occupant? "being piloted by [key_name(MT.chassis.occupant)]." : "without a pilot."] ")
+			message_admins("[key_name(usr)][ADMIN_FLW(usr)] has activated remote EMP on exosuit [MT.chassis][ADMIN_JMP(MT.chassis)], which is currently [MT.chassis.occupant? "being piloted by [key_name(MT.chassis.occupant)][ADMIN_FLW(MT.chassis.occupant)]." : "without a pilot."] ")
```
Comment (Jordie0608) [inline code/game/mecha/mecha_control_console.dm]: Use `key_name_admin` for `message_admins`

**Review by Jordie0608 — CHANGES_REQUESTED** [review]

File: code/game/mecha/mecha_control_console.dm (hunk 2 above)
Comment (zxaber) [reply code/game/mecha/mecha_control_console.dm]: Done and done

File: code/game/mecha/mecha_control_console.dm [hunk 3]
```diff
@@ -55,6 +57,7 @@
 	w_class = WEIGHT_CLASS_SMALL
 	var/ai_beacon = FALSE //If this beacon allows for AI control. Exists to avoid using istype() on checking.
 	var/recharging = 0
+	var/obj/mecha/chassis
```
Comment (ninjanomnom) [inline code/game/mecha/mecha_control_console.dm]: You need to null this in `Destroy()`

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/game/mecha/mecha_control_console.dm (hunk 3 above)
Comment (zxaber) [reply code/game/mecha/mecha_control_console.dm]: done

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**